index.d.ts

```diff
@@ -33,7 +33,7 @@ export type LimitFunction = {
 	@returns A Promise that returns an array of results.
 	*/
 	map: <Input, ReturnType> (
-		array: readonly Input[],
+		array: Iterable<Input> | ArrayLike<Input>,
```

Comment:
```suggestion
		array: Iterable<Input>,
```

I don't want to support array-like objects.